docs/sources/flow/reference/config-blocks/remotecfg.md

```diff
@@ -0,0 +1,92 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/reference/config-blocks/remotecfg/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/config-blocks/remotecfg/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/config-blocks/remotecfg/
+- /docs/grafana-cloud/send-data/agent/flow/reference/config-blocks/remotecfg/
+canonical: remotecfgs://grafana.com/docs/agent/latest/flow/reference/config-blocks/remotecfg/
+description: Learn about the remotecfg configuration block
+menuTitle: remotecfg
+title: remotecfg block
+---
+
+# remotecfg block
+
+`remotecfg` is an optional configuration block that enables {{< param "PRODUCT_NAME" >}}
+to fetch and load configuration from a remote endpoint.
+`remotecfg` is specified without a label and can only be provided once per
+configuration file.
+
+The [API definition][] for managing and fetching configuration that the
+`remotecfg` block uses is available under the Apache 2.0 license.
+
+[API definition]: https://github.com/grafana/agent-remote-config
+
+## Example
+
+```river
+remotecfg {
+	url = "SERVICE_URL"
+	basic_auth {
+		username      = USERNAME
+		password_file = PASSWORD_FILE
+	}
+
+	id             = constants.hostname
+	metadata       = {"cluster" = "dev-us-central-0", "namespace" = "agent-otlp"}
+	poll_frequency = "10m"
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name             | Type                 | Description                                      | Default     | Required
+-----------------|----------------------|--------------------------------------------------|-------------|---------
+`url`            | `string`             | The address of the API to poll for configuration | `""`        | no
+`id`             | `string`             | A self-reported ID                               | `see below` | no
+`metadata`       | `map(string)`        | A set of self-reported metadata                  | `{}`        | no
+`poll_frequency` | `duration`           | How often to poll the API for new configuration  | `"1m"`      | no
+
+If not set, the self-reported `id` that the Agent uses is a randomly generated,
+anonymous unique ID (UUID) that is stored on the Agent's storage path as
+`agent_seed.json` so that it can persist across restarts.
```

Comment:
```suggestion
anonymous unique ID (UUID) that is stored on t{{< param "PRODUCT_ROOT_NAME" >}}'s storage path as
`agent_seed.json` so that it can persist across restarts.
```